Add new commands

lukeeey opened the pull request:
This pull request aims to add many commands from vanilla Minecraft and have the syntax and messages as close as possible. **Im really sorry that i had to make it 3 times, i messed up with git**

### Changes
* Added `EntityLiving.hasEffects()` to return whether or not an entity has active effects
* Added `EntityPlayer.setOp`
* Added `EntityPlayer.sendActionbar`
* Added `EntityPlayer.sendTitleAnimationTimes`
* Renamed a few constants in the `PotionEffect` enum to make it easier for the /effect command
* Added `PotionEffect.getEffectName()` to return the human readable name from the constant
* Added `Gamemode.getGamemodeName()` to return the human readable name
* Added `World.getDifficulty()`
* Added `World.setDifficulty( Difficulty difficulty )`
* Changed `EntityPlayer.set/getLevel()` to `set/getXPLevel()`

### Commands
- [x] `/op <player>`
- [x] `/deop <player>`
- [x] `/title`
  - [x] `/title <player> clear`
  - [x] `/title <player> reset`
  - [x] `/title <player> <title|subtitle|actionbar> <text>`
  - [x] `/title <player> times <fadeIn> <duration> <fadeOut>`
- [x] `/effect`
  - [x] `/effect <player> clear`
  - [x] `/effect <player> <effect> [seconds] [amplifier] [hideParticles]`
- [x] `/gamemode <mode> [player]` 
- [x] `/difficulty <difficulty>` 
- [x] `/list`   (changed to be exactly like vanilla)
- [ ] `/give <player> <item> [amount] [data] [components]` 
- [x] `/tell <target> <message>`
- [ ] `/setblock`
- [ ] `/xp <amount> [player]` (basic implementation but need postfixes)
- [ ] `/stopsound [player] [sound]` (seems to be missing a parameter? needs looking in to)
- [ ] `/playsound [sound] [player] [pitch] [volume] [minimumVolume]` (overloads seem to be in the wrong order)

### Follow up
* Remove gamemode name and potion effect name from the API?

lukeeey committed:
Lets try this again

lukeeey committed:
Fix build errors

Janmm14 commented on gomint-server/src/main/java/io/gomint/server/command/vanilla/DifficultyCommand.java:
```diff
@@ -0,0 +1,71 @@
+
+        Difficulty difficulty;
+        switch( String.valueOf( arguments.get( "difficulty" ) ) ) {
+			case "0":
```
indention

Janmm14 commented on gomint-server/src/main/java/io/gomint/server/command/vanilla/GamemodeCommand.java:
```diff
@@ -0,0 +1,93 @@
+
+        Gamemode mode;
+        switch( String.valueOf( arguments.get( "mode" ) ) ) {
+			case "0":
```
indention

lukeeey replied:
Im not worried about that right now because it wont even load, and that indentation is correct in InteliJ

Janmm14 replied:
The problem is that spaces and tabs are mixed in this file. Because github doesn#T display tabs as 4 spaces such code style issues are visible.

Commit:
Remove GOMINT world support (#1)

Add 1.7.0.2 support (currently only spawns, need a better entity data flag system for the rest)
Parse the material numbers from the JSON files
Fix leveldb world loading a little (its still unuseable with new worlds)

lukeeey committed:
Merge upstream
